### ISSUE TITLE
Atmosphere GWT sample

### ISSUE THREAD

**Issue by ternite** [Atmosphere/atmosphere-samples#36]
Hi,

I'm trying to run the gwt20-managed-rpc sample to understand how to integrate atmosphere into my GWT application.

According to README.md (https://github.com/Atmosphere/atmosphere-samples/blob/master/README.md), I have to compile the project, then execute "mvn jetty:run". This works fine, but what after that? I visited the running webserver under http://localhost:8080/ and there actually is a page titled "GWT RPC demo for atmosphere" with an empty text box and nothing else (no buttons).

How do I trigger messages? According to samples.client.GwtRpcDemo.java, there should be a Button added to a div named "buttonbar", but apparently, nothing happens:

![2014-10-26_1838](https://cloud.githubusercontent.com/assets/7860777/4783817/12c1d7de-5d37-11e4-8246-f8c978630e80.png)

How do I test if there is a service made ready on the server side? Is there any "bootstrap" code that instantiates and initializes samples.server.ManagedGWTResource? When trying to integrate the code into my application, I finally had some success using:

```java
Config.Builder builder = new Config.Builder();
builder.resource(ManagedGWTResource.class);
...
```

I cannot find such a construct in gwt20-managed-rpc.

I could successfully run the gwt20-websockets sample (as described in https://github.com/Atmosphere/atmosphere-samples/blob/master/extensions-samples/gwt/gwt20-websockets/README.md).

But I am not interested in implementing an only websocket implementation. I want to use a ManagedService that transparently manages the message protocols used.

After mentioning my tries to integrate the code into my project, I might as well tell about the problems that made me try troubleshoot the sample again.

When running my custom implementation, the call to "subsribe" on the Atmosphere object results in the following error (on client side): "Breaking on exception: TypeError: Cannot read property 'subscribe' of undefined"

Here is the respective call in my application, which is directly derivated from the call in the client sample code in samples.client.GwtRpcDemo.java:

```java
Atmosphere atmosphere = Atmosphere.create();
final AtmosphereRequest jerseyRpcRequest = atmosphere.subscribe(jerseyRpcRequestConfig);
```

The error happens within subscribe(...) while calling org.atmosphere.gwt20.client.Atmosphere:subscribeImpl(RequestConfig requestConfig) which has a native implementation. 

I'm not sure (since I don't get debugging running in gwt20-managed-rpc), but I think it's likely that this particular call to subscribe is successfully traversed in my running instance of gwt20-managed-rpc. At least, onModuleLoad() should be called and no error arises. I can only assume that there must be a version conflict in my configuration so that the native implementation cannot be found in the compiled code because it might have been renamed.

Here is the dependency excerpt from my pom.xml:

```xml
<dependency>
	<groupId>com.google.gwt</groupId>
	<artifactId>gwt-servlet</artifactId>
	<version>2.7.0-beta1</version>
	<scope>runtime</scope>
</dependency>
<dependency>
	<groupId>com.google.gwt</groupId>
	<artifactId>gwt-user</artifactId>
	<version>2.7.0-beta1</version>
	<scope>provided</scope>
</dependency>
<dependency>
	<groupId>org.atmosphere.extensions</groupId>
	<artifactId>atmosphere-gwt20-client</artifactId>
	<version>2.3.0-SNAPSHOT</version>
	<scope>provided</scope>
</dependency>
<dependency>
	<groupId>org.atmosphere</groupId>
	<artifactId>atmosphere-runtime</artifactId>
	<version>2.2.3</version>
</dependency>
<dependency>
	<groupId>org.atmosphere</groupId>
	<artifactId>nettosphere</artifactId>
	<version>2.1.1</version>
</dependency>
```

(Note that I am using 2.3.0-SNAPSHOT of atmosphere-gwt20-client. I cloned the project and compiled it myself, as compatibility with version 2.7.0-beta1 of GWT was integrated there only 3 days ago.)

Best regards
ternite

**Comment by isairuiz** [comment]
i'm have the same error in my maven application using atmophere jquery idon't know why.

**Comment by jfarcand** [comment]
OK tested with 2.3.0 and every things works as expected. Please let me know if you still have issue.

**Comment by BenDol** [comment]
I am using 2.3.0-GWT27 with 2.3.0 runtime and still having this problem.
```xml
            <!-- Atmosphere -->
            <dependency>
                <groupId>org.atmosphere</groupId>
                <artifactId>atmosphere-runtime</artifactId>
                <version>2.3.0</version>
            </dependency>
            <dependency>
                <groupId>org.atmosphere.extensions</groupId>
                <artifactId>atmosphere-gwt20-client</artifactId>
                <version>2.3.0-GWT27</version>
            </dependency>
            <dependency>
                <groupId>org.atmosphere.extensions</groupId>
                <artifactId>atmosphere-gwt20-server</artifactId>
                <version>2.3.0-GWT27</version>
            </dependency>
            <dependency>
                <groupId>org.atmosphere.extensions</groupId>
                <artifactId>atmosphere-gwt20-jackson</artifactId>
                <version>2.3.0-GWT27</version>
            </dependency>
```
Getting this error:
```
Sat May 09 11:05:49 GMT+1200 2015 nz.co.doltech.actions.client.Application$PreApplicationImpl
SEVERE: (TypeError) : Cannot read property 'subscribe' of undefinedcom.google.gwt.core.client.JavaScriptException: (TypeError) : Cannot read property 'subscribe' of undefined
	at Unknown.wrap_2_g$(Exceptions.java:36)
	at Unknown.$subscribeImpl_0_g$(Atmosphere.java:52)
	at Unknown.$subscribe_0_g$(Atmosphere.java:38)
```

**Comment by knutmoen** [comment]
Did you solve your problem? I seem to have the same problem as you had.

**Comment by ternite** [comment]
No, I did not. I was trying to learn about Atmosphere and quit doing so after not being able to apply it in my usage scenario. In case you would want to know my alternative: I did not find one. But I admit I was not intensely looking for it, as it was for educational purposes only.

**Comment by TotalFluke** [comment]
I'm also getting this exact same problem, I've tried a range of versions of atmosphere-gwt20-client and it happens in all of them.

Any ideas what else I can try to fix this?

**Comment by TotalFluke** [comment]
I managed to solve my issue. The javascript for atmosphere was missing from the war directory. I used the following maven dependency...
```xml	
<dependency>
	<groupId>org.atmosphere.client</groupId>
	<artifactId>javascript</artifactId>
	<version>2.3.1</version>
	<type>war</type>
	<scope>compile</scope>
</dependency>
```
But this did not actually place the javascript is a place where it could be found by my project. I'm not sure it placed it anywhere at all. This is probably because I'm not using Maven correctly. Instead I had to manually add this javascript to the project.

Also, I needed to manually include the javascript in the html file like so...

```html
    <script type="text/javascript" language="javascript" src="javascript/atmosphere.js"></script>
```

This fixed the problems described in this thread.
